recipes/rocksdb/build.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+set -eu
+
+### Create Makefiles
+cmake -DCMAKE_PREFIX_PATH=$PREFIX \
+      -DCMAKE_INSTALL_PREFIX=$PREFIX \
+      -DCMAKE_INSTALL_LIBDIR=lib \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DFAIL_ON_WARNINGS=ON \
+      -DPORTABLE=ON \
+      -DUSE_RTTI=ON \
+      -DWITH_GFLAGS=ON \
+      -DWITH_JEMALLOC=ON \
+      -DWITH_LZ4=OFF \
+      -DWITH_SNAPPY=ON \
+      -DWITH_TESTS=OFF \
+      -DWITH_TOOLS=OFF \
+      -DWITH_ZLIB=OFF \
```

Comment:
Not against this, but any specific reason why BZ2, LZ4, ZLIB, ZSTD are turned off?

Reply:
I am building RocksDB to support building cockroach from source. That  version only depended on snappy, that is why I built it with this configuration.

If we turn the other backends on, then I think we should build multiple versions of rocksdb and use a _mutex package to select the right backend. Is that how you would do it as well?  If so, let me know and I will add a placeholder _rocksdb_mutex package.

Reply:
From this wiki page: https://github.com/facebook/rocksdb/wiki/Compression
It seems that the choice is dynamic. It defaults to snappy and one can override it. So I don't think a mutex is required here, since it is not exactly mutually exclusive in that sense.

Reply:
Awesome, I will do a build with LZ4 (and maybe ZTD) and see if cockroachdb can compile and run. If it does, I will leave it like that.

Reply:
I tried with cockroach locally and it worked. I am adding LZ4 and ZTD as I said before. 

It seem overkill to add BZ2 and ZLIB since we have better performing compression backends.